examples/simpleGov/README.md

```diff
@@ -0,0 +1,1064 @@
+
+```
+x
+└─── simple_governance
+      ├─── client
+      │     ├───  cli
+      │     │     └─── simple_governance.go
+      │     └─── rest
+      │           └─── simple_governance.go
+      ├─── errors.go
+      ├─── handler_test.go
+      ├─── handler.go
+      ├─── keeper_keys.go
+      ├─── keeper_test.go
+      ├─── keeper.go
+      ├─── test_common.go
+      ├─── types_test.go
+      ├─── types.go
+      └─── wire.go
+```
+
+Let us go into the detail of each of these files.
+
+### Types
+
+In this file, we define the custom types for our module. This includes the types from the [State](#State) section and the custom message types defined in the [Transactions](#Transactions) section.
+
+For each new type that is not a message, it is possible to add methods that make sense in the context of the application. In our case, we will implement an `updateTally` function to easily update the tally of a given proposal as vote messages come in.
+
+Messages are a bit different. They implement the `Message` interface defined in the SDK's `types` folder. Here are the methods you need to implement when you define a custom message type:
+
+- `Type()`: This function returns the name of our module's route. When messages are processed by the application, they are routed using the string returned by the `Type()` method.
+- `GetSignBytes()`: Returns the byte representation of the message. It is used to sign the message.
+- `GetSigners()`: Returns address(es) of the signer(s).
+- `ValidateBasic()`: This function is used to discard obviously invalid messages. It is called at the beginning of `runTx()` in the baseapp file. If `ValidateBasic()` does not return `nil`, the app stops running the transaction.
+- `Get()`: A basic getter, returns some property of the message.
+- `String()`: Returns a human-readable version of the message
+
+For our simple governance messages, this means:
+
+- `Type()` will return `"simpleGov"`
+- For `SubmitProposalMsg`, we need to make sure that the attributes are not empty and that the deposit is both valid and positive. Note that this is only basic validation, we will therefore not check in this method that the sender has sufficient funds to pay for the deposit
+- For `VoteMsg`, we check that the address and option are valid and that the proposalID is not negative.
+- As for other methods, less customization is required. You can check the code to see a standard way of implementing these.
+
+### Keeper
+
+#### Short intro to keepers
+
+`Keepers` handles read and writes for modules' stores. This is where the notion of capability enters into play.
+
+As module developers, we have to define keepers to interact with our module's store(s) not only from within our module, but also from other modules. When another module wants to access one of our module's store(s), a keeper for this store has to be passed to it at the application level. In practice, it will look like that:
+
+```go
+// in app.go
```

Comment:
Where is app.go? this is a file we haven't mentioned yet.

Reply:
I reworked the tuto a bit so that `app.go` shows up earlier